pallets/pallet-dip-provider/src/lib.rs

```diff
@@ -31,23 +31,17 @@ pub mod pallet {
 	use frame_support::{pallet_prelude::*, traits::EnsureOrigin};
 	use frame_system::pallet_prelude::*;
 	use parity_scale_codec::FullCodec;
-	use sp_io::MultiRemovalResults;
 	use sp_std::fmt::Debug;
 
-	use crate::traits::{IdentityCommitmentGenerator, IdentityProvider, SubmitterInfo};
+	use crate::traits::{IdentityCommitmentGenerator, IdentityProvider, ProviderHooks, SubmitterInfo};
 
+	pub type IdentityProviderOf<T> = <T as Config>::IdentityProvider;
 	pub type IdentityOf<T> = <<T as Config>::IdentityProvider as IdentityProvider<<T as Config>::Identifier>>::Success;
 	pub type IdentityCommitmentVersion = u16;
```

Comment:
For my learning: Why not create a type in the config that implements the default trait? Wouldn't it be cleaner? I often struggle to determine what should be included in the configuration and what should not. :confused:

Reply:
You mean using the latest Substrate features that allow to specify defaults for a pallet's `Config` trait? Or what do you mean?

Reply:
I took a shot in https://github.com/KILTprotocol/kilt-node/pull/577. Resolving this, let me know there what you think!